### DISCUSSION TITLE
API of Interactive Brokers

### DISCUSSION THREAD

**Discussion by TrendoBoom** [aaguiar10/gflows#9] [Q&A]
I tried to put IP:PORT of my API,
But didn't work well could you show me the right way to pull data from Interactive Brokers

**Comment by TrendoBoom** [comment]
![image](https://github.com/user-attachments/assets/a26bd087-953b-4ce6-aefd-56effcca14fb)

**Comment by TrendoBoom** [comment]
![image](https://github.com/user-attachments/assets/415e9d4b-fbf1-4128-9b8b-d0655f77f26e)

without env its work, what's wrong ?

**Comment by aaguiar10** [comment]
Hi @TrendoBoom,

What tickers are you passing in?

Can you print the output of the ticker_info variable in **layout.py**. This is within the serve_layout function

```
ticker_info = Ticker(tickers).quote_type
print(ticker_info)
```

**Reply by TrendoBoom** [reply]
I use IBKR API and the ticker SPX

**Reply by TrendoBoom** [reply]
> Hi @TrendoBoom,
> 
> What tickers are you passing in?
> 
> Can you print the output of the ticker_info variable in **layout.py**. This is within the serve_layout function
> 
> ```
> ticker_info = Ticker(tickers).quote_type
> print(ticker_info)
> ```

How to get the output ?

**Reply by aaguiar10** [reply]
> > Hi @TrendoBoom,
> 
> > 
> 
> > What tickers are you passing in?
> 
> > 
> 
> > Can you print the output of the ticker_info variable in **layout.py**. This is within the serve_layout function
> 
> > 
> 
> > ```
> 
> > ticker_info = Ticker(tickers).quote_type
> 
> > print(ticker_info)
> 
> > ```
> 
> 
> 
> How to get the output ?

Add the print(ticker_info) statement under the ticker_info line in that function, then run the program with python app.py. It should print the output to your command line before the error happens

**Reply by TrendoBoom** [reply]
![image](https://github.com/user-attachments/assets/ede29ab0-153a-419b-9873-f14e47ff0bac)
![image](https://github.com/user-attachments/assets/4900cf78-7f8c-419d-a788-2fdc6a3b9d42)

**Reply by TrendoBoom** [reply]
how I chose SPX of CBOE

![image](https://github.com/user-attachments/assets/ffa9101a-c963-4202-8df3-e159cc050f63)

**Reply by aaguiar10** [reply]
Thanks,

The ticker should be ^SPX (with the ^) since that's the format Yahoo Finance uses for indexes like the S&P 500, Nasdaq 100, etc.

**Reply by TrendoBoom** [reply]
![image](https://github.com/user-attachments/assets/561871eb-13e6-4d7b-9d32-f5468fe33695)
![image](https://github.com/user-attachments/assets/2e623b58-d1f9-4737-9bd7-f21e2b27c01d)

**Reply by TrendoBoom** [reply]
> Thanks,
> 
> The ticker should be ^SPX (with the ^) since that's the format Yahoo Finance uses for indexes like the S&P 500, Nasdaq 100, etc.

how to ...

**Reply by aaguiar10** [reply]
Wherever you inputted the ticker (assuming it's in the environment variable TICKERS) ![image](https://github.com/user-attachments/assets/f6166748-1f12-4e05-a789-a6fe1f0725ee)

**Reply by TrendoBoom** [reply]
![image](https://github.com/user-attachments/assets/3794e5ff-ad71-405b-988d-0d4699127266)

**Reply by TrendoBoom** [reply]
how to make it work ?

**Reply by aaguiar10** [reply]
Add an http:// in front of the url you've inputted in API_URL

**Reply by TrendoBoom** [reply]
![image](https://github.com/user-attachments/assets/1e622420-f402-4563-b738-9aab10f17f50)

I tried http:// and https:// but its take more than half seems frozen hour any solution

**Reply by aaguiar10** [reply]
> ![image](https://private-user-images.githubusercontent.com/20939954/390940364-1e622420-f402-4563-b738-9aab10f17f50.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MzAzNDYsIm5iZiI6MTczMjkzMDA0NiwicGF0aCI6Ii8yMDkzOTk1NC8zOTA5NDAzNjQtMWU2MjI0MjAtZjQwMi00NTYzLWI3MzgtOWFhYjEwZjE3ZjUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTMwVDAxMjcyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJkZTk2NzY1Yjc3ZGM5MjJiMjg3NTNhYmE4NTU1YWZhMjM2NjI5NzFmMGY5ZTM1YTE5NmMxYjg5MDllY2E4ZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JZzjmjAuj2GJITsrB2qTSPA2d5uCPiDqvt6hgK37Hfg)
> 
> I tried http:// and https:// but its take more than half seems frozen hour any solution

@TrendoBoom,

Integrating IB can be a bit complex because it requires a solid understanding of their API. Since this is outside the scope of the project, I'd recommend checking out their documentation for more detailed guidance on setting up the connection and continuing from there. Using print() statements can be helpful for debugging and tracking the connection.
